stackit/internal/services/serverbackup/schedule/resource.go

```diff
@@ -0,0 +1,539 @@
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(1, 255),
+				},
+			},
+			"backup_schedule_id": schema.Int64Attribute{
+				Description: "Backup schedule ID",
+				Computed:    true,
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.UseStateForUnknown(),
+				},
+				Validators: []validator.Int64{
+					int64validator.AtLeast(1),
+				},
+			},
+			"project_id": schema.StringAttribute{
+				Description: "STACKIT Project ID to which the server is associated.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+			},
+			"server_id": schema.StringAttribute{
+				Description: "Server ID for the backup schedule",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+			},
+			"rrule": schema.StringAttribute{
+				Description: "Backup schedule described in `rrule` (recurrence rule) format.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Validators: []validator.String{
+					validate.Rrule(),
+					validate.NoSeparator(),
+				},
+			},
+			"enabled": schema.BoolAttribute{
+				Description: "Is the backup schedule enabled or disabled.",
+				Required:    true,
+			},
+			"backup_properties": schema.ObjectAttribute{
+				Description: "Backup schedule details for the backups",
+				Optional:    true,
+				AttributeTypes: map[string]attr.Type{
+					"volume_ids": types.ListType{
+						ElemType: types.StringType,
+					},
+					"name":             types.StringType,
+					"retention_period": types.Int64Type,
+				},
+			},
+		},
+	}
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *scheduleResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) { // nolint:gocritic // function signature required by Terraform
+	var model scheduleResourceModel
+	diags := req.Plan.Get(ctx, &model)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	projectId := model.ProjectId.ValueString()
+	serverId := model.ServerId.ValueString()
+	ctx = tflog.SetField(ctx, "project_id", projectId)
+	ctx = tflog.SetField(ctx, "server_id", serverId)
+
+	// Enable backups if not already enabled
+	err := enableBackupsService(ctx, &model, r.client)
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error creating schedule", fmt.Sprintf("Enabling serverbackup project before creation: %v", err))
```

Comment:
```suggestion
		core.LogAndAddError(ctx, &resp.Diagnostics, "Error creating server backup schedule", fmt.Sprintf("Enabling serverbackup project before creation: %v", err))
```

Reply:
This is applicable everywhere in the file, to avoid confusions with other possible backups existing in the provider